### DISCUSSION TITLE
RRF compatibility

### DISCUSSION THREAD

**Discussion by bttguy** [bigtreetech/BIGTREETECH-TouchScreenFirmware#1763] [Ideas]
Just opening a discussion for #1400 for easier communication/testing/FR ideas for RRF users. If you have any idea/sollution and want to discuss it before a FR/PR, post it here, and we can help you in testing/feedback.

**Comment by bttguy** [comment]
A basic WIP documentation for RRF users

TFT Screen Configuration and support for RRF

Overview
The most recent version of the standard bigtreetech TFT firmware has built in support for RepRapFirmware. The pre-built images have this enabled by default.

Loading the firmware
There is now an RRF config.ini
It needs to be renamed from config_rrf.ini to config.ini for flashing of the firmware.

Config.g Changes
Add the following line to your config.g to enable the screen: M575 P1 S2 B57600

Start.g Changes
Add the following line in your start.g file to allow the screen to know when a job has started.
M409 K"job.file.fileName"

Stop.g Changes
Add the following line in your stop.g file to allow the screen to know when a job has stopped.
M118 P2 S"//action:cancel"
M0 also need to be added to your stop gcode in your slicer to allow stop.g to be ran.

Pause.g Changes
Add the following line in your pause.g file to allow the screen to know when a job has been aused.
M118 P2 S"//action::paused"

Cancel.g Changes
Add the following line in your cancel.g file to allow the screen to know when a job has been cancelled.
M118 P2 S"//action::prompt_begin Resuming"

Implemented features: 
- Auto detect fw type + separate config.ini for easier setup
                        -Temperature/movement/cooling status monitoring and control
			- Print status monitoring with mid-print tuneing/pausing
			- Macro support
			- Print from onboard/external SD card

Feature requests in #1791:

**Reply by jaysuk** [reply]
Don't forget you need ```M0``` in your stop gcode in your slicer for stop.g to be ran
